.github/workflows/sketch-compile.yml

```diff
@@ -0,0 +1,53 @@
+name: Sketch Compile
+
+# the action is triggered at every pull request
+on:
+  - pull_request
+
+jobs:
+  sketch-compile:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        fqbn:
+          - arduino:samd:mkrwifi1010
+          - arduino:samd:nano_33_iot
+          - arduino:megaavr:uno2018:mode=on
+          - arduino:mbed:nano33ble
+          - arduino:mbed_nano:nanorp2040connect
+          - arduino:renesas_uno:unor4wifi
+          - "esp8266:esp822:d1_mini"
+
+    steps:
+      # This step makes the contents of the repository available to the workflow
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # For more information: https://github.com/arduino/compile-sketches#readme
+      - name: Compile sketch
+        uses: arduino/compile-sketches@v1
+        with:
+          # The default is to compile for the Arduino Uno board. If you want to compile for other boards, use the `fqbn` input.
+          sketch-paths: |
+            # search the sketches in the current repo (no subfolders)
+            - ./
+
+          platforms: |
+            - name: "arduino:samd"
+            - name: "arduino:megaavr"
+            - name: "arduino:mbed"
+            - name: "arduino:mbed_nano"
+            - name: "arduino:renesas_uno"
+            - source-url: "https://github.com/esp8266/Arduino/releases/download/3.1.2/esp8266-3.1.2.zip"
+              name: "esp8266:d1_mini"
+              source-path: "./esp8266-3.1.2"
+
+          libraries: |
+            - name: time
+            - name: Wire
+            - name: MFRC522
+            - name: ESP8266WiFi
+            - name: ESP8266WebServer
+            - name: WiFiClient
+            - name: WiFiClientSecure
+            - name: LiquidCrystal_I2C
```

Comment:
here I tried to grab all the imports we have in the various scripts like:
```
import < ESP8266WiFi .h>
```
In the example provided [here](https://blog.arduino.cc/2021/04/09/test-your-arduino-projects-with-github-actions/) it seems the author is using the `Arduino_` prefix like:
```
- name: Arduino_ ESP8266WiFi
- ...
```